src/components/AssetsPage/index.jsx

```diff
@@ -44,63 +37,41 @@ export class AssetsPage extends React.Component {
     }
   }
 
-  hasNoAssets = assetsList => (
-    assetsList.length === 0
-  )
+  getPageType = (props) => {
+    const numberOfAssets = props.assetsList.length;
 
-  hasAssets = assetsList => (
-    assetsList.length > 0
-  )
-
-  handleCheckBoxChange = (checked) => {
-    this.setState({
-      checked,
-    });
+    if ('type' in props.status && props.status.type === assetActions.request.REQUESTING_ASSETS) {
```

Comment:
I have added a REQUESTING_ASSETS action. This represents the time between making an API call and getting the response. This was added to prevent the page type from changing in between those two periods. The logic here is that, if we've made a request and are waiting on the response, do not change the page view until the response is received, so that user's experience is not interrupted.